lib/ansible/plugins/strategy/linear.py

```diff
@@ -211,30 +211,21 @@ def run(self, iterator, play_context):
                                 skip_rest = True
                                 break
 
-                        run_once = templar.template(task.run_once) or action and getattr(action, 'BYPASS_HOST_LOOP', False)
+                        run_once = action and getattr(action, 'BYPASS_HOST_LOOP', False) or templar.template(task.run_once)
+                        try:
+                            task.name = to_text(templar.template(task.name, fail_on_undefined=False), nonstring='empty')
+                        except Exception as e:
+                            display.debug(f"Failed to templalte task name ({task.name}), ignoring error and continuing: {e}")
 
                         if (task.any_errors_fatal or run_once) and not task.ignore_errors:
                             any_errors_fatal = True
 
                         if not callback_sent:
-                            display.debug("sending task start callback, copying the task so we can template it temporarily")
-                            saved_name = task.name
-                            display.debug("done copying, going to template now")
-                            try:
-                                task.name = to_text(templar.template(task.name, fail_on_undefined=False), nonstring='empty')
-                                display.debug("done templating")
-                            except Exception:
-                                # just ignore any errors during task name templating,
-                                # we don't care if it just shows the raw name
-                                display.debug("templating failed for some reason")
-                            display.debug("here goes the callback...")
                             if isinstance(task, Handler):
                                 self._tqm.send_callback('v2_playbook_on_handler_task_start', task)
                             else:
                                 self._tqm.send_callback('v2_playbook_on_task_start', task, is_conditional=False)
-                            task.name = saved_name
```

Comment:
I am wondering why we revert to the original untemplated name. It appears to have been introduced in https://github.com/ansible/ansible/commit/c0dfa8d5121ee3588efc4b036880b25488b6fbb8#diff-db48a56313ab62e2be5bfaa24dbc894b74c0c88342a4232b74c5a978b58b8d82R195 by templating on a copy of the task but without any information as to why it is needed.

My guess was that a task name could contain a host specific template but we show the templated task name only once for all hosts in lockstep so it does not matter.

For the record we do not revert to the original task name in the `free` strategy.

Reply:
that was my conclusion also, the code was being defensive but we really don't need it